modules/tinymce/src/core/main/ts/keyboard/SpaceKey.ts

```diff
@@ -8,17 +8,22 @@
 import Editor from '../api/Editor';
 import { EditorEvent } from '../api/util/EventDispatcher';
 import VK from '../api/util/VK';
-import { fireFakeInputEvent } from './FakeInputEvents';
+import { fireFakeBeforeInputEvent, fireFakeInputEvent } from './FakeInputEvents';
 import * as InsertSpace from './InsertSpace';
 import * as MatchKeys from './MatchKeys';
 
 const executeKeydownOverride = (editor: Editor, evt: KeyboardEvent) => {
-  MatchKeys.execute([
+  MatchKeys.executeWithDelayedAction([
     { keyCode: VK.SPACEBAR, action: MatchKeys.action(InsertSpace.insertSpaceOrNbspAtSelection, editor) }
-  ], evt).each(() => {
+  ], evt).each((apply) => {
```

Comment:
For consistency can you please call this `applyAction` or change your other version to just be `apply`?